### ISSUE TITLE
Miner stops reading some disks if a new block comes in before all drives are read.

### ISSUE THREAD

**Issue by Leozolt** [Creepsky/creepMiner#346]
When a new block is announced while the miner is reading the disks, some drives (the drives that did not finish yet) do not get read in the next blocks. If it happens very early in the read process more drives are left out on the next blocks. (The drives that did not finish seem to be left out in the next blocks and the miner doesn't recover from it so the problem gets worse over time until no drives are read at all)

Release 1.7.16

[mining.conf.txt](https://github.com/Creepsky/creepMiner/files/1815733/mining.conf.txt)
[creepMiner_20180315_140131.165603.log](https://github.com/Creepsky/creepMiner/files/1815753/creepMiner_20180315_140131.165603.log)

Using Ext4, ntfs and exFat

maybe the issue is similar to #177

**Comment by loppefaaret** [comment]
yeah, your "/media/leozolt/Burst1" does not finish after round 818 - perhaps verify that the drive is not dying ?

**Comment by Leozolt** [comment]
Yes restarting the miner helps until a fast block comes in before the reading finishes again. Removed ../Burst1 from the config & need some fast block now to verify. It's not only ../Burst1 but it's usually the first drive to fail being read, as it is always the slowest 10TB exFat. Ok without ../Burst1 it is ../Burst3 failing first (which is the second exFat). Trying to also remove ../Burst3 now

**Comment by Leozolt** [comment]
The miner doesn't hang on fast blocks without the exFat drives. Any idea what could cause this behavior (Ubuntu 16.04 with exfat-fuse 1.2.3-1)

**Comment by Leozolt** [comment]
For the people that like screenshots: the instance in focus always shows 92,39%, updates the block info but does not read the exfat disks.
![creepsky-2instances-exfat-hangs](https://user-images.githubusercontent.com/26518970/37541630-8cc6d27a-295b-11e8-9511-4987257684d5.jpg)

**Comment by Leozolt** [comment]
maybe exfat-fuse driver needs a special "reset current read-command"? That is higher than my skill level sadly.. I agree that it could possibly be a machine specific or host system specific issue but maybe soemone here has a hint how to solve this (whitout replotting on ext4)((I would need to truncate the plots t make them fit by copying on the same size ext4))

**Comment by Leozolt** [comment]
Partially resolved the failure by running all non exFat drives in one (server) instance and running the 2 other drives using exFat on individual instances for each drive. First 1.7.16-2 reads, then 1.7.16-3 reads. no failure.  Weird 
![separate-instances-mixed-drives](https://user-images.githubusercontent.com/26518970/37546207-083b3b98-296c-11e8-834d-aa4512267a37.jpg)

**Comment by Leozolt** [comment]
Maybe it is something with exFat driver not supporting more than x threads from one instance?

**Comment by loppefaaret** [comment]
when the disks fail to complete the read, do you have any kernel messages about read errors?

**Comment by Leozolt** [comment]
working on the filesystem is abnormally slow, got a touch test pass through after a fast block came in an the miner stuck on 11% read (some disks got read after) but no ls later on the drive i probed
.

**Comment by kattycris** [comment]
i have the same problem, i test miner 1.7 and 1.8 and not fix.....
